File: tasks/0020-valid-parentheses.js
"use strict";

/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  for (let i = 0; i < arr.length; ) {
    if (
      s.indexOf("()") === -1 &&
      s.indexOf("{}") === -1 &&
      s.indexOf("[]") === -1
    )
      break;
    if (s.indexOf("()") !== -1)
      s = s.slice(0, s.indexOf("()")) + s.slice(s.indexOf("()") + 2, s.length);
    if (s.indexOf("{}") !== -1)
      s = s.slice(0, s.indexOf("{}")) + s.slice(s.indexOf("{}") + 2, s.length);
    if (s.indexOf("[]") !== -1)
      s = s.slice(0, s.indexOf("[]")) + s.slice(s.indexOf("[]") + 2, s.length);
  }
  return s.length === 0;
};

File: tasks/0020-valid-parentheses.go
func isValid(s string) bool {
  stack := []string{}

  for _, v := range s {
    switch string(v) {
      case "(":
        stack = append(stack, "ao")
      case ")":
        if len(stack) == 0 || stack[len(stack) - 1] != "ao" {
          return false
        } else {
          stack = stack[:len(stack) - 1]
        }
      case "{":
        stack = append(stack, "bo")
      case "}":
        if len(stack) == 0 || stack[len(stack) - 1] != "bo" {
          return false
        } else {
          stack = stack[:len(stack) - 1]
        }
      case "[":
        stack = append(stack, "co")
      case "]":
        if len(stack) == 0 || stack[len(stack) - 1] != "co" {
          return false
        } else {
          stack = stack[:len(stack) - 1]
        }
    }
  }

  return len(stack) == 0
}